vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,88 @@ For more canonical documentation of defining packages dependency check the offic
 
 Some of internally used C routines are now exported on C level thus can be used in R packages directly from their C code. See [`?cdt`](https://rdatatable.gitlab.io/data.table/reference/cdt.html) for details and [Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html) _Linking to native routines in other packages_ section for usage.
 
-## Importing from non-R Applications {non-r-API}
+## Importing from non-r Applications {non-r-api}
 
 Some tiny parts of `data.table` C code were isolated from the R C API and can now be used from non-R applications by linking to .so / .dll files. More concrete details about this will be provided later; for now you can study the C code that was isolated from the R C API in [src/fread.c](https://github.com/Rdatatable/data.table/blob/master/src/fread.c) and [src/fwrite.c](https://github.com/Rdatatable/data.table/blob/master/src/fwrite.c).
+
+## How to convert your Depends dependency on data.table to Imports
+
+To convert a `Depends` dependency on `data.table` to an `Imports` dependency in an R package, follow these steps:
+
+### 1. Update the DESCRIPTION File
+In your package's `DESCRIPTION` file, change the entry from `Depends` to `Imports`. This involves moving `data.table` from the `Depends` field to the `Imports` field.
+
+**Before:**
+```r
+Depends:
+    R (>= 3.5.0),
+    data.table
+Imports:
+```
+
+**After:**
+```r
+Depends:
+    R (>= 3.5.0)
+Imports:
+    data.table
+```
+
+### 2. Modify the NAMESPACE File
+Ensure that any functions from `data.table` that you use in your package are imported. This can be done either by explicitly importing specific functions or by importing the entire package but specific imports are considered best practice.
+
+#### Importing Specific Functions
+If you use specific functions like `data.table()`, `setDT()`, or `merge()`, you can import them individually in the `NAMESPACE` file:
+
+```r
+importFrom(data.table, data.table, setDT, merge)
+```
+
+#### Bulk Import
+Alternatively, you can import all functions from `data.table` at once, though this is generally not recommended as best practice:
```

Comment:
It will be good to justify this a bit. Here are my top 2 reasons:

 1. The NAMESPACE can serve as good documentation of _how_ you depend on certain packages
 2. Blanket `import()` leaves you open to subtle breakage, e.g. you `import(pkgA); import(pkgB)`, but later `pkgB` exports a function also exported by `pkgA`. This will break your package because it will cause a conflict in your namespace which is disallowed by `R CMD check` (and hence CRAN).

There might be more reasons you can add. I just don't like reading "it's not best practice" claims without any hint at _why_.